### PR TITLE
`atan2()` - New CSS functional notation

### PR DESCRIPTION
**Pull request by ramiy**
### Description
Document the CSS `atan2()` functional notation.

### Motivation
[Safari/webkit already supports](https://webkit.org/blog/12445/new-webkit-features-in-safari-15-4/) this CSS function, [Firefox is shipping](https://bugzilla.mozilla.org/show_bug.cgi?id=1774589) this too. [Chromium](https://bugs.chromium.org/p/chromium/issues/detail?id=1190444) will follow soon.

### Additional details
https://drafts.csswg.org/css-values/#trig-funcs

#### Metadata
- [x] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

### CODE REVIEW

**Comment by bsmth** [comment]
Thanks @ramiy , I will take a look shortly 👍🏻

File: files/en-us/web/css/atan2/index.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  - Reference
+  - Web
+  - atan2
+  - Experimental
+browser-compat: css.types.atan2
+spec-urls: https://drafts.csswg.org/css-values/#trig-funcs
+---
+{{CSSRef}}{{SeeCompatTable}}
+
+The **`atan2()`** [CSS](/en-US/docs/Web/CSS) [function](/en-US/docs/Web/CSS/CSS_Functions) is a trigonometric function that returns the inverse tangent of a number between `-infinity` and `infinity`. The function contains a calculation that returns the number of radians representing an {{cssxref("&lt;angle&gt;")}} between `-180deg` and `180deg`.
```
Comment (bsmth) [inline files/en-us/web/css/atan2/index.md:18]:
```suggestion
The **`atan2()`** [CSS](/en-US/docs/Web/CSS) [function](/en-US/docs/Web/CSS/CSS_Functions) is a trigonometric function that returns the inverse tangent of two values between `-infinity` and `infinity`. The function accepts two arguments and returns the number of radians representing an {{cssxref("&lt;angle&gt;")}} between `-180deg` and `180deg`.
```

File: files/en-us/web/css/atan2/index.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+transform: rotate(atan2(15, 90));
+
+/* Other values */
+transform: rotate(atan2(pi, 45));
+transform: rotate(atan2(e, 30));
+```
+
+### Parameter
+
+The `atan2(y, x)` function accepts two comma-separated value as its parameters.
```
Comment (bsmth) [inline files/en-us/web/css/atan2/index.md:34]: I would mention at this point what the values that `x` and `y` can be, explaining that they can be as described below, but both values must be of the same type.

```suggestion
The `atan2(y, x)` function accepts two comma-separated values as its parameters.
The values can be a {{cssxref("&lt;number&gt;")}}, {{cssxref("&lt;dimension&gt;")}} or {{cssxref("&lt;percentage>&gt;")}}, but both must be the same type.
```

Comment (bsmth) [reply files/en-us/web/css/atan2/index.md:34]: I would also add:

>Given two points `x` and `y`, `atan2(y, x) calculates the angle between the positive x axis and the ray from the origin to the point (x, y).

File: files/en-us/web/css/atan2/index.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+The **`atan2()`** [CSS](/en-US/docs/Web/CSS) [function](/en-US/docs/Web/CSS/CSS_Functions) is a trigonometric function that returns the inverse tangent of two values between `-infinity` and `infinity`. The function accepts two arguments and returns the number of radians representing an {{cssxref("&lt;angle&gt;")}} between `-180deg` and `180deg`.
+
+## Syntax
+
+```css
+/* Single <number> values */
+transform: rotate(atan2(90, 15));
+transform: rotate(atan2(15, 90));
+
```
Comment (bsmth) [inline files/en-us/web/css/atan2/index.md:26]: Can we add some examples for `<dimension>` and `<percentage>` also?

File: files/en-us/web/css/atan2/index.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+- `y`
+  - : The y coordinate of the point. A {{cssxref("&lt;number&gt;")}}, {{cssxref("&lt;dimension&gt;")}} or {{cssxref("&lt;percentage>&gt;")}}.
+
+- `x`
+  - : The y coordinate of the point. A {{cssxref("&lt;number&gt;")}}, {{cssxref("&lt;dimension&gt;")}} or {{cssxref("&lt;percentage>&gt;")}}.
+
+### Return value
+
+The function returns the {{cssxref("&lt;angle&gt;")}} between the positive X-axis and the point (x,y).
+
```
Comment (bsmth) [inline files/en-us/web/css/atan2/index.md:46]: Shall we add some examples like the other pages? There's a pen here that shows the different types:

https://codepen.io/bsmth/pen/gOzwrjo

File: files/en-us/web/css/atan2/index.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+/* Single <number> values */
+transform: rotate(atan2(90, 15));
+transform: rotate(atan2(15, 90));
+
+/* Other values */
+transform: rotate(atan2(pi, 45));
+transform: rotate(atan2(e, 30));
+```
+
+### Parameter
```
Comment (bsmth) [inline files/en-us/web/css/atan2/index.md:32]:
```suggestion
### Parameters
```

File: files/en-us/web/css/atan2/index.md (hunk 4 above)
Comment (ramiy) [reply files/en-us/web/css/atan2/index.md:45]: I'm using Firefox, with the trig functions flag enabled, but the `atan2()` doesn't rotate the boxes...

Comment (bsmth) [reply files/en-us/web/css/atan2/index.md:50]: You might need a later version, I'm using `106.0a1 (2022-09-10)` from nightly. It also works in Safari (`Version 15.6.1 (17613.3.9.1.16)` for me.

Comment (bsmth) [reply files/en-us/web/css/atan2/index.md:50]: It's also working on the [105.0b9 Beta](https://www.mozilla.org/en-US/firefox/channel/desktop/#beta) but it's behind the trig pref (`layout.css.trig.enabled`) like you mentioned.

File: files/en-us/web/css/atan2/index.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+  - : The y coordinate of the point. A {{cssxref("&lt;number&gt;")}}, {{cssxref("&lt;dimension&gt;")}} or {{cssxref("&lt;percentage&gt;")}}.
+
+### Return value
+
+Given two points `x` and `y`, the function `atan2(y, x)` calculates and returns the {{cssxref("&lt;angle&gt;")}} between the positive `x` axis and the ray from the origin to the point `(x, y)`.
+
+### Formal syntax
+
+{{CSSSyntax}}
+
```
Comment (bsmth) [inline files/en-us/web/css/atan2/index.md:54]:
```suggestion
## Example

The `atan2()` function can be used to {{cssxref("transform-function/rotate", "rotate")}} elements as it return an {{cssxref("&lt;angle&gt;")}}.

```html hidden
<div class="box box-1"></div>
<div class="box box-2"></div>
<div class="box box-3"></div>
<div class="box box-4"></div>
<div class="box box-5"></div>
```

```css hidden
body {
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 50px;
}
```

```css
div.box {
  width: 100px;
  height: 100px;
  background: linear-gradient(orange, red);
}
div.box-1 {
  transform: rotate(atan2(3, 2));
}
div.box-2 {
  transform: rotate(atan2(3%, -2%));
}
div.box-3 {
  transform: rotate(atan2(-1, 0.5));
}
div.box-4 {
  transform: rotate(atan2(1, 0.5));
}
div.box-5 {
  transform: rotate(atan2(1rem, -0.5rem));
}
```

{{EmbedLiveSample('example', '100%', '200px')}}

```

**Review by bsmth — APPROVED** [review]
Thanks @ramiy 👍🏻

**Pull request merged**